lang/scala/scala.py

```diff
@@ -0,0 +1,269 @@
+        actions.auto_insert(" < ")
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(" <= ")
+
+    def code_operator_and():
+        actions.auto_insert(" && ")
+
+    def code_operator_or():
+        actions.auto_insert(" || ")
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(" & ")
+
+    def code_operator_bitwise_and_assignment():
+        actions.auto_insert(' &= ')
+
+    def code_operator_increment():
+        actions.auto_insert('++')
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(" | ")
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(" ^ ")
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(" << ")
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(" <<= ")
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(" >> ")
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(" >>= ")
+
+    def code_self():
+        actions.auto_insert("this")
+
+    def code_insert_null():
+        actions.auto_insert("null")
+
+    def code_insert_is_null():
+        actions.auto_insert(" == null")
+
+    def code_insert_is_not_null():
+        actions.auto_insert(" != null")
+
+    def code_state_if():
+        actions.insert("if () ")
+        actions.edit.left()
+        actions.edit.left()
+
+    def code_state_else_if():
+        actions.insert("else if () ")
+        actions.edit.left()
+        actions.edit.left()
+
+    def code_state_else():
+        actions.insert("else ")
+        actions.key("enter")
+
+    def code_state_switch():
+        actions.insert("match {\n")
+
+    def code_state_case():
+        actions.insert("case  => ")
+        actions.edit.left()
+        actions.edit.left()
+        actions.edit.left()
+        actions.edit.left()
+
+    def code_state_for():
+        actions.insert("for () ")
+        actions.edit.left()
+        actions.edit.left()
+
+    def code_state_while():
+        actions.insert("while () ")
+        actions.edit.left()
+        actions.edit.left()
+
+    def code_break():
+        actions.auto_insert('break')
+
+    def code_next():
+        actions.auto_insert('continue')
+
+    def code_insert_true():
+        actions.auto_insert('true')
+
+    def code_insert_false():
+        actions.auto_insert('false')
+
+    def code_define_class():
+        actions.auto_insert("class ")
+
+    def code_import():
+        actions.auto_insert("import ")
+
+    def code_state_return():
+        actions.insert("return ")
+
+    def code_comment_line_prefix():
+        actions.auto_insert('// ')
+
+    def code_block_comment():
+        actions.insert('/*')
+        actions.key('enter')
+        actions.key('enter')
+        actions.insert('*/')
+        actions.edit.up()
+
+    def code_block_comment_prefix():
+        actions.auto_insert('/*')
+
+    def code_block_comment_suffix():
```

Comment:
code_block_comment_suffix -> code_comment_block_suffix

Reply:
did these change? I definitely tested each of these when I wrote them the first time weeks ago?

Reply:
Yeah, I think they did along with the big refactoring of the language features.